jdaviz/core/data_formats.py

```diff
@@ -0,0 +1,115 @@
+                                                readwrite='Read')
+
+file_to_config_mapping = {i: default_mapping.get(
+    i, 'specviz') for i in formats_table['Format']}
+
+# default n-dimension to configuration mapping
+ndim_to_config_mapping = {1: 'specviz', 2: 'specviz2d', 3: 'cubeviz'}
+
+
+def guess_dimensionality(filename):
+    """ Guess the dimensionality of a file
+
+    Parameters
+    ----------
+    filename : str
+        The filename of the loaded data
+
+    Returns
+    -------
+    ndim : int
+        The number of dimensions of the data
+    """
+
+    # check for valid string input
+    if not isinstance(filename, (str, pathlib.Path)) or not os.path.isfile(filename):
+        raise ValueError(f'{filename} is not a valid string path to a file')
+
+    # get the dimension number of the first extension
+    ndim = astropy.io.fits.getval(filename, 'NAXIS', ext=1)
+    return ndim
+
+
+def get_valid_format(filename):
+    """ Identify a best match jdaviz configuration from a filename
+
+    Parameters
+    ----------
+    filename : str
+        The filename of the loaded data
+
+    Returns
+    -------
+        valid_format : str
+            A valid file format
+        config : str
+            The recommended application configuration
+    """
+
+    valid_file_format = identify_spectrum_format(filename)
+    ndim = guess_dimensionality(filename)
+
+    if valid_file_format:
+        recommended_config = file_to_config_mapping.get(valid_file_format, 'default')
+    else:
+        recommended_config = ndim_to_config_mapping.get(ndim, 'default')
+
+    return valid_file_format, recommended_config
+
+
+def identify_data(filename, current=None):
+    """ Identify the data format and application configuration from a filename
+
+    Parameters
+    ----------
+    filename : str
+        The filename of the loaded data
+    current : str
+        The currently loading application configuration, if any
+
+    Returns
+    -------
+        valid_format : str
+            A valid file format
+        config : str
+            The recommended application configuration
+        status : str
+            A status message
+    """
+
+    valid_format, config = get_valid_format(filename)
+
+    # get a list of pre-built configurations
+    valid_configs = list_configurations()
+
+    # - check that valid_format exists
+    # - check that config is in the list of available configurations
+    # - check that config does not conflict with the existing configuration
+
+    if not valid_format:
+        status = 'Error: Cannot determine format of the file to load.  Please specify a format'
+    elif config not in valid_configs:
+        status = f"Error: Config {config} not a valid configuration."
+    elif current and config != current:
+        status = 'Error: Mismatch between input file format and loaded configuration.'
+    else:
+        status = 'Success: Valid Format'
+
+    return valid_format, config, status
```

Comment:
I'm not clear why a status message is needed here given this is sort of a "library function".  Perhaps instead this:
```suggestion
    if not valid_format:
        raise ValueError('Cannot determine format of the file to load.  Please specify a format')
    elif config not in valid_configs:
        raise ValueError(f"Config {config} not a valid configuration.")
    elif current and config != current:
        raise ValueError('Mismatch between input file format and loaded configuration.')

    return valid_format, config
```
and then the "user" function in MAST instead uses ``try/except`` to catch the error cases?  ala
```
try:
    valid_format, config = identiy_data(...)
    status = 'Success: Valid Format'
except ValueError as e:
    status = 'Error:' + e.args[0]
```